fxdc/config.py

```
@@ -37,17 +54,88 @@ def __init__(
+                raise NullFailure(f"Field {key} cannot be None")
+
         if self.from_data:
-            return self.from_data(*args, **kwargs)
-        return self.class_(*args, **kwargs)
+            return self.from_data(*args, **newkwargs)
+        return self.class_(*args, **newkwargs)
 
     def __repr__(self) -> str:
         return self.classname
 
     def return_data(self, obj: object) -> dict[str, Any]:
```

Ruff (0.12.2)

`return_data` is too complex (11 > 10)

(C901)